sks_cluster: add support for auto-upgrades

falzm opened the pull request:
This changes introduces a new `auto_upgrade` attribute to the
`exoscale_sks_cluster` resource to control automatic control plane
Kubernetes version upgrading.

marcaurele commented on exoscale/compute_resource.go:
```diff
@@ -256,6 +262,9 @@ func createCompute(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
+	if username == "" {
+		username = getSSHUsername(image)
```
Does this default to `root` ?

marcaurele replied:
You should get rid of `getSSHUsername` and default to `root` in case the username is not in the template details.

greut replied:
I'll keep it for the `readCompute` part of the story though.

greut commented:
:rocket: thanks for the review!

Commit:
sks_cluster: add support for auto-upgrades

This changes introduces a new `auto_upgrade` attribute to the
`exoscale_sks_cluster` resource to control automatic control plane
Kubernetes version upgrading.
